### PR TITLE
Style edits for OIDC warning

### PR DESCRIPTION
**Pull request by Jessie212**
## Description

> Is this change a fix, improvement, new feature, refactoring, or other?

Improvement.

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

Changes made to documentation.

> How would you describe this change to a non-technical end user or system administrator?

Edit warning in OIDC section.

## Documentation

( ) No documentation is needed.
(x) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: docs/src/main/sphinx/security/oauth2.rst [hunk 1]
```diff
@@ -41,12 +41,12 @@ values to set corresponding OAuth2 authentication configuration properties:
...
-  check the validity of OAuth2 access tokens. This is unnecessary as JWTs can be
-  inspected locally and using them against ``userinfo_endpoint`` may even result
-  in authentication failure. In this case set:
-  ``http-server.authentication.oauth2.oidc.use-userinfo-endpoint=false`` which
+  If the authorization server is issuing JSON Web Tokens (JWTs), and the
+  metadata document contains ``userinfo_endpoint``, Trino uses this endpoint
+  to check the validity of OAuth2 access tokens. Since JWTs can be inspected
+  locally, using them against ``userinfo_endpoint`` may result in authentication
+  failure. In this case, set ``userinfo_endpoint`` to ``false`` (
+  ``http-server.authentication.oauth2.oidc.use-userinfo-endpoint=false``). This
```
Comment (jhlodin) [inline docs/src/main/sphinx/security/oauth2.rst:49]: Reword to "In this case, set the ``http-server.authentication.oauth2.oidc.use-userinfo-endpoint`` configuration property to ``false``.

File: docs/src/main/sphinx/security/oauth2.rst [hunk 2]
```diff
@@ -41,12 +41,12 @@ values to set corresponding OAuth2 authentication configuration properties:
 
 .. warning::
 
-  In situation when Authorization Server is issuing JSON Web Tokens (JWTs) but the
-  metadata document contains ``userinfo_endpoint`` Trino will use this endpoint to
-  check the validity of OAuth2 access tokens. This is unnecessary as JWTs can be
-  inspected locally and using them against ``userinfo_endpoint`` may even result
-  in authentication failure. In this case set:
-  ``http-server.authentication.oauth2.oidc.use-userinfo-endpoint=false`` which
+  If the authorization server is issuing JSON Web Tokens (JWTs), and the
```
Comment (jhlodin) [inline docs/src/main/sphinx/security/oauth2.rst:44]: Remove this first comma

**Review by jhlodin — APPROVED** [review]
LGTM % minor edits

**Review by mosabua — APPROVED** [review]
Looks good. Thanks

**Review by lukasz-walkiewicz — APPROVED** [review]

**Pull request merged**